### ISSUE TITLE
current_page? with trailing_slash

### ISSUE THREAD

**Issue by araslanov-e** [rails/rails#19472]
`GET "/products/"`
```
current_page?(products_path trailing_slash: true) # => false
```

**Comment by eileencodes** [comment]
We need more information before we can help you. Closing for now, I'll reopen when you send more information. Please let us know what issue you are having and what Rails version you're using.

**Comment by araslanov-e** [comment]
Rails 4.2.0
Method `current_page?` not work with `Rails.application.routes.url_helpers` with option `trailing_slash: true`

**Comment by anthonycollini** [comment]
+1. 

if `config.action_controller.default_url_options = { trailing_slash: true }` is set in application.rb

`request.env['PATH_INFO']` will reply with `/plans'

and therefore `/plans` is not equal to `/plans/`

**Comment by eileencodes** [comment]
Can someone who is having this issue create a small application or script that demonstrates this issue?

**Comment by tgxworld** [comment]
To replicate the issue, just generate a new application and submit a GET request and add a trailing slash. Just add a debugger in either the views or the controller's action and you'll see that `request.path_info` does not contain the trailing slash.

Trailing slash is removed in https://github.com/rails/rails/blob/master/actionpack/lib/action_dispatch/middleware/static.rb#L108

**Comment by tgxworld** [comment]
This has not been fixed.

**Comment by maclover7** [comment]
New PR has been opened to solve this by @st0012, --> #25520
